x-pack/plugins/infra/public/components/waffle/node.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiToolTip } from '@elastic/eui';
+import { darken } from 'polished';
+import React from 'react';
+import styled from 'styled-components';
+import { InfraWaffleMapNode, InfraWaffleOptions } from '../../lib/lib';
+import { NodeContextMenu } from './node_context_menu';
+
+const initialState = {
+  isPopoverOpen: false,
+};
+
+type State = Readonly<typeof initialState>;
+
+interface Props {
+  onDrilldown: () => void;
+  squareSize: number;
+  options: InfraWaffleOptions;
+  node: InfraWaffleMapNode;
+  formatter: (val: number) => string;
+}
+
+export class Node extends React.PureComponent<Props, State> {
+  public readonly state: State = initialState;
+  public render() {
+    const { node, formatter, options, squareSize } = this.props;
+    const { isPopoverOpen } = this.state;
+    const metric = node.metrics.find(m => m.name === 'count');
+    const valueMode = squareSize > 75;
+    const label = 'Count';
+    const value = metric != null ? formatter(metric.value) : 'n/a';
+
+    return (
+      <NodeContextMenu
+        node={node}
+        isPopoverOpen={isPopoverOpen}
+        closePopover={this.closePopover}
+        options={options}
+      >
+        <EuiToolTip position="top" content={`${node.name} | ${value}`}>
+          <Container style={{ width: squareSize, height: squareSize }} onClick={this.togglePopover}>
+            <SquareOuter>
+              <SquareInner>
+                {valueMode && (
+                  <ValueInner>
+                    <Label>{label}</Label>
+                    <Value>{value}</Value>
+                  </ValueInner>
+                )}
+              </SquareInner>
+            </SquareOuter>
+          </Container>
+        </EuiToolTip>
+      </NodeContextMenu>
+    );
+  }
+
+  private togglePopover = () => {
+    this.setState(prevState => ({ isPopoverOpen: !prevState.isPopoverOpen }));
+  };
+
+  private closePopover = () => {
+    this.setState({ isPopoverOpen: false });
+  };
+}
+
+export const Container = styled.div`
```

Comment:
Could we use a less generic name for this?

Reply:
See above